geoprocessing/src/modules/files/files.service.ts

```diff
@@ -2,32 +2,41 @@ import { Injectable } from '@nestjs/common';
 import { createReadStream } from 'fs';
 import { Extract } from 'unzipper';
 import * as path from 'path';
-import { unlink, rmdir } from 'fs/promises';
+import { unlink, rmdir, readdir } from 'fs/promises';
 
 @Injectable()
 export class FileService {
-  unzipFile(fileInfo: Express.Multer.File): Promise<string> {
+  unzipFile(
+    origin: string,
+    fileName: string,
+    destination: string,
+  ): Promise<string> {
     return new Promise((resolve, reject) => {
-      createReadStream(fileInfo.path)
+      createReadStream(origin)
         .pipe(
           Extract({
             path: path.join(
-              fileInfo.destination,
-              path.basename(fileInfo.filename.replace('.zip', '')),
+              destination,
+              path.basename(fileName.replace('.zip', '')),
             ),
           }),
         )
-        .on('close', () =>
-          resolve(`${fileInfo.filename} extracted successfully`),
-        )
+        .on('close', () => resolve(`${fileName} extracted successfully`))
         .on('error', (error: Error) =>
-          reject(
-            new Error(`${fileInfo.filename} could not be extracted: ` + error),
-          ),
+          reject(new Error(`${fileName} could not be extracted: ` + error)),
         );
     });
   }
 
+  async areFilesInFolder(
+    path: string,
+    fileExtensions: Array<string>,
+  ): Promise<boolean> {
+    const filesInPath = await readdir(path);
+    const extensions = filesInPath.map((file) => file.split('.').pop());
+    return fileExtensions.every((ext: string) => extensions.includes(ext));
+  }
```

Comment:
the method lacks cohesion to the class

Reply:
@Dyostiq could you explain it a little bit please?

Reply:
Thoughts and opinions always appreciated, thanks @hotzevzl a0793bab77839c74ad79ca3231d99a45f4078167